exonum-java-binding-core/rust/src/storage/proof_map_index.rs

```diff
@@ -442,11 +442,10 @@ pub extern "system" fn Java_com_exonum_binding_storage_indices_ProofMapIndexProx
 }
 
 fn convert_to_key(env: &JNIEnv, array: jbyteArray) -> JniResult<Key> {
-    // TODO: Optimize copying and allocations.
-    let bytes = env.convert_byte_array(array)?;
-    assert_eq!(PROOF_MAP_KEY_SIZE, bytes.len());
+    let array_length = env.get_array_length(array)?;
+    assert_eq!(PROOF_MAP_KEY_SIZE as jint, array_length);
```

Comment:
Shouldn't we return error instead of panic?

Reply:
A panic will be intercepted and converted into a Java Error.
I suppose, if here is a Java array of a wrong length, things are totally broken and the application should be terminated.